lsp-mode.el

```diff
@@ -3984,6 +3984,15 @@ REFERENCES? t when METHOD returns references."
     (signal 'lsp-capability-not-supported (list "definitionProvider")))
   (lsp-find-locations "textDocument/definition" nil :display-action display-action))
 
+(defun lsp-find-definition-mouse (click)
+  "Click to start `lsp-find-definition' at clicked point."
+  (interactive "e")
+  (let ((p1 (posn-point (event-start click))))
+    (goto-char p1)
+    (lsp-find-definition)))
+
+(define-key lsp-mode-map (kbd "C-<down-mouse-1>") #'lsp-find-definition-mouse)
```

Comment:
Can you group that in the initialization option, like that: 

```
(defvar tide-mode-map
  (let ((map (make-sparse-keymap)))
    (define-key map (kbd "M-.") #'tide-jump-to-definition)
    (define-key map (kbd "M-,") #'tide-jump-back)
    map))
```
AFAIK this is the standart pattern to initialize a modemap.